clients/banking/src/components/MerchantProfilePaymentArea.tsx

```diff
@@ -0,0 +1,435 @@
+  buttonText: string;
+};
+
+type EmptyListWithCtaProps = {
+  merchantProfile: MerchantPaymentsQuery["merchantProfile"];
+  action: Action;
+};
+
+const EmptyListWithCta = ({ merchantProfile, action }: EmptyListWithCtaProps) => {
+  const { actionType, buttonText, handleAction, subtitle, title } = action;
+
+  return (
+    <Box justifyContent="center">
+      {actionType === "canCreatePayment" &&
+        (isNotNullish(merchantProfile?.merchantLogoUrl) ? (
+          <AutoWidthImage
+            alt={merchantProfile.merchantName}
+            height={40}
+            maxWidth={180}
+            resizeMode="contain"
+            sourceUri={merchantProfile.merchantLogoUrl}
+            style={styles.centered}
+          />
+        ) : (
+          <LakeHeading variant="h3" level={3} align="center">
+            {merchantProfile?.merchantName}
+          </LakeHeading>
+        ))}
+
+      {match(actionType)
+        .with("shouldEnablePaymentMethod", () => (
+          <Box alignItems="center">
+            <BorderedIcon name="lake-merchant" />
+          </Box>
+        ))
+        .with("waitingForReview", () => (
+          <Box alignItems="center">
+            <BorderedIcon name="lake-clock" />
+          </Box>
+        ))
+        .otherwise(() => null)}
+
+      <Space height={24} />
+
+      <LakeText variant="medium" align="center" color={colors.gray[900]}>
+        {title}
+      </LakeText>
+
+      <Space height={4} />
+
+      <LakeText align="center" variant="smallRegular">
+        {subtitle}
+      </LakeText>
+
+      <Space height={24} />
+
+      <Box alignItems="center">
+        <LakeButton
+          color="current"
+          onPress={handleAction}
+          disabled={actionType === "waitingForReview"}
+        >
+          {buttonText}
+        </LakeButton>
+      </Box>
+    </Box>
+  );
+};
+
+type Props = {
+  params: GetRouteParams<"AccountMerchantsProfilePaymentsArea">;
+  large: boolean;
+};
+
+export const MerchantProfilePaymentArea = ({ params, large }: Props) => {
+  const { merchantProfileId, accountMembershipId } = params;
+
+  const { canCreateMerchantPaymentLinks } = usePermissions();
+
+  const route = Router.useRoute([
+    "AccountMerchantsProfilePaymentsList",
+    "AccountMerchantsProfilePaymentsDetails",
+  ]);
+
+  const filters: MerchantPaymentFilters = useMemo(() => {
+    return {
+      paymentMethod: isNotNullish(params.paymentMethod)
+        ? Array.filterMap(params.paymentMethod, item =>
+            match(item)
+              .with("Card", "Check", "DirectDebit", value => Option.Some(value))
+              .otherwise(() => Option.None()),
+          )
+        : undefined,
+      status: isNotNullish(params.status)
+        ? Array.filterMap(params.status, item =>
+            match(item)
+              .with("Authorized", "Captured", "Initiated", "Rejected", item => Option.Some(item))
+              .otherwise(() => Option.None()),
+          )
+        : undefined,
+    } as const;
```

Comment:
Note that you can use `Option`, or `ts-pattern` `isMatching` here to simplify the logic:

```ts
const filters: MerchantPaymentFilters = useMemo(() => {
  return {
    paymentMethod: Option.fromNullable(params.paymentMethod)
      .map(methods => methods.filter(isMatching(P.union("Card", "Check", "DirectDebit"))))
      .toUndefined(),

    status: Option.fromNullable(params.status)
      .map(status =>
        status.filter(isMatching(P.union("Authorized", "Captured", "Initiated", "Rejected"))),
      )
      .toUndefined(),
  } as const;
}, [params.paymentMethod, params.status]);
```